tests/health_checks/NDv4/test-bandwidth.sh

```diff
@@ -35,13 +35,251 @@ catch_error() {
+    	    cudaEventCreate(&start);
+    	    cudaEventCreate(&stop);
+            double totaltime = 0.0;
+    	    float localtime = 0.0;
+    	    for(int iloop=0;iloop<warmup+loops;++iloop)
+    	    {
+    	        cudaEventRecord(start,0);
+    		if(htod){gpuErrorCheck(cudaMemcpyAsync(gpuBuffer[gpu],\
+				hostBuffer[numa],size,cudaMemcpyDefault));}
+    		else{gpuErrorCheck(cudaMemcpyAsync(hostBuffer[numa],\
+				gpuBuffer[gpu],size,cudaMemcpyDefault));}
+    	        cudaEventRecord(stop,0);
+    	        cudaEventSynchronize(stop);
+    
+    	        dummy_kernel<<<1, 64>>>(gpuBuffer[gpu]);
+    
+    	        cudaEventElapsedTime(&localtime, start, stop);
+    	        if(iloop>=warmup){totaltime += localtime;}
+    	    }
+    	    if(time){
+            double m1 = double(loops)/sizeof(uint8_t);
+	    double m2 = double(numEntries)/1.0e6;
+    	    printf("gpu%d_numa%d %f \n",gpu,numa,m1*m2/totaltime);
+    	    }
+    	    cudaDeviceSynchronize();
+        }
+    }
+}
+
+
+int main(int argc, char** argv)
+{
+    int ishtod=-1;
+    size_t numEs = 0;
+    if(argc > 1)
+    {
+	for(int j=1;j<argc;++j)
+	{
+            int cmphtod = strcmp(argv[j],"--htod");
+            int cmpdtoh = strcmp(argv[j],"--dtoh");
+            if(cmphtod == 0){ishtod=1;}
+            if(cmpdtoh == 0){ishtod=0;}
+
+	    int cmpsize = strcmp(argv[j],"--size");
+	    if(cmpsize == 0)
+	    {
+                j+=1;
+		if(argc-1<j)
+		{
+	            printf("Missing an input to --size <number> \n");
+		    exit(1);
+		}
+		char * ptr;
+		long a = strtol(argv[j],&ptr,10);
+
+	        if ((ptr == argv[j]) || (*ptr != '\0'))
+		{
+		    printf("Size input was not read correctly \n");
+		    exit(1);
+		}
+		numEs = size_t(a);
+	    }
+
+	}
+    }
+    if((ishtod!=0) && (ishtod!=1))
+    {
+        printf("Unrecognized option. Please specify --htod or --dtoh.\n");
+        return 1;
+    }
+    if((numEs==0))
+    {
+        printf("Must specify a size with --size <number>\n");
+        return 1;
+    }
+		    
+    // Get number of NUMA nodes
+    if (numa_available()) 
+    {
+        fprintf(stderr, "main::numa_available error\n");
+	printf("numa_available error failed");
+        return -1;
+    }
+    int num_numa = numa_num_configured_nodes();
+    int num_gpus = 0;
+    gpuErrorCheck(cudaGetDeviceCount(&num_gpus));
+    //const size_t numEntries=16777216*4;
+    const size_t numEntries=numEs;
+
+    uint8_t** hostBuffer = new uint8_t *[num_numa];
+    uint8_t** gpuBuffer = new uint8_t *[num_gpus];
+
+    initialize(num_numa,num_gpus,numEntries,gpuBuffer,hostBuffer);
+
+    const int warmup=10;
+    const int loops=20;
+
+    copy_loop(num_gpus, num_numa, warmup, 0, numEntries, 0, gpuBuffer,\
+		    hostBuffer, ishtod);
+    copy_loop(num_gpus, num_numa, warmup, loops, numEntries, 1, gpuBuffer,\
+		    hostBuffer, ishtod);
+
+    cleanup(num_numa,num_gpus,numEntries,gpuBuffer,hostBuffer);
+}
+EOF
+
+compile_bench="nvcc -lnuma bandwidth.cu -o gpu-copy"
```

Comment:
During setup we currently build superbench's bandwidth test. If we have built that test this script will run it instead of compiling this bandwidth test, otherwise it will build and run this one. They can take the same inputs and produce similar bandwidth values. 

But good point, I should probably modify the startup so we are always building the same test.

Reply:
Yea, its just weird that its being built as part of the test each time. One time is enough.

Reply:
It will only be built if the executable is not already present. The code tests if it exists before calling the function that writes and compiles the executable.